src/vm.c

```diff
@@ -483,6 +483,26 @@ int tjs__load_file(JSContext *ctx, DynBuf *dbuf, const char *filename) {
     return r;
 }
 
+JSValue TJS_EvalModuleContent(JSContext *ctx, const char *filename, bool is_main, bool use_real_path, const char *content, size_t len) {
+    /* Compile then run to be able to set import.meta */
+    JSValue ret = JS_Eval(ctx, content, len, filename, JS_EVAL_TYPE_MODULE | JS_EVAL_FLAG_COMPILE_ONLY);
+    if (!JS_IsException(ret)) {
+        js_module_set_import_meta(ctx, ret, use_real_path, is_main);
+        ret = JS_EvalFunction(ctx, ret);
+    }
+
+    /* Emit window 'load' event. */
```

Comment:
Just moved some code from below, since it is executed when loading a worker from a file, so I guess that should also be the case if the content is in a variable?